File: WebServer/app/models/device.rb
class Device < ActiveRecord::Base
	belongs_to :mib
  has_many :wall_raports
  validates :name, :length => {:maximum => 256}, :presence => true
  validates :description, :length => {:maximum => 1024}, :presence => true
  validates :room, :presence => true, :length => {:maximum => 25}
  validates :mac, :presence => true, :uniqueness => true
  validates :model, :presence => true
  validates :peername, :presence => true
  validates :port, :presence => true
  validates :login, :presence => true
  validates :password, :presence => true
end

File: init_device/model/device.cpp
#ifndef __DEVICE__
#define __DEVICE__

#include <string>

class Device {
private:
	int id;
	std::string name;
	std::string description;
	std::string room;
	std::string mac;
	std::string serial_number;
	std::string model;
	std::string peername;
	int port_number;
	//authorization
	std::string login;
	std::string password;
	// mib
	int mib_id;
	
	//helper for deamon
	int status;

public: 
	Device() 
	{
	}

	int getId()
	{
		return id;
	}

	void setId(int id)
	{
		this->id = id;
	}
	
	std::string getName()
	{
		return name;
	}
	
	void setName(std::string name)
	{
		this-> name = name;
	}
	
	std::string getLogin()
	{
		return login;
	}
	
	void setPeername(std::string peername)
	{
		this->peername = peername;
	}

	std::string getPeername()
	{
		return peername;
	}

	void setLogin(std::string login)
	{
		this -> login = login;
	}
		
	std::string getPassword()
	{
		return password;
	}

	void setPassword(std::string password) 
	{
		this -> password = password;
	}

	int getStatus()
	{
		return status;
	}
	
	void setStatus(int status)
	{
		this -> status = status;
	}
	
};

#endif

